platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +102,168 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
     @Override
     public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
+
+        // now that the DelegateServletFilter has been initialized with a filterConfig, initialize all SecurityFilters using that filterConfig
+        for (ServiceReference<SecurityFilter> securityFilterServiceReference : sortedSecurityFilterServiceReferences) {
+            initFilter(securityFilterServiceReference, filterConfig);
+        }
+
+        filtersHaveBeenInitialized = true;
+    }
+
+    public void addSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference)
+            throws ServletException {
+        LOGGER.trace("Adding SecurityFilter ranking={} name={}",
+                securityFilterServiceReference.getProperty(Constants.SERVICE_RANKING),
+                ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+
+        if (filtersHaveBeenInitialized) {
```

Comment:
❓ Is there a reason that you've chosen only to init the filters here in the case where the `init` method has already been called? Wouldn't it work the same and simplify everything if you just dumped this boolean flag and _always_ called `initFilter` in this method and never looped and called it in the `init` method?